### ISSUE TITLE
Cannot read property 'printString' of undefined

### ISSUE THREAD

**Issue by juniorschroder** [januslo/react-native-sunmi-inner-printer#32]
device: https://sunmi.com/en/v1.html

- npm module installed on project with --save
- linked with success
- Removed @Override of public List<Class<? extends JavaScriptModule>> createJSModules() in SunmiInnerPrinterPackaje.java to prevent compile errors

So, my app was build with success, but when i'm call printString this error happens.
debuggin my app:
import SunmiInnerPrinter from 'react-native-sunmi-inner-printer';
console.log(SunmiInnerPrinter) - this is undefined.

Anyone can help?

**Comment by juniorschroder** [comment]
Thnx for quick response.

Seems like don't have any typo issue.

A have Linked with success after instal npm package, check files settings.gradle, app/build.gradle it's ok, MainPackage.java doesn't exists, maybe MainApplication.java? 
in MainApplication.java it's OK too:
package com.newerppdvmobile;

import android.app.Application;

import com.facebook.react.ReactApplication;
import com.sunmi.innerprinter.SunmiInnerPrinterPackage;
import com.oblador.vectoricons.VectorIconsPackage;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;

import java.util.Arrays;
import java.util.List;

public class MainApplication extends Application implements ReactApplication {

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
            new SunmiInnerPrinterPackage(),
            new VectorIconsPackage()
      );
    }

    @Override
    protected String getJSMainModuleName() {
      return "index";
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
  }
}



here is my App.js:

/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View
} from 'react-native';
import { Button, Header } from 'react-native-elements';
import SunmiInnerPrinter from 'react-native-sunmi-inner-printer';

const instructions = Platform.select({
  ios: 'Press Cmd+R to reload,\n' +
    'Cmd+D or shake for dev menu',
  android: 'Double tap R on your keyboard to reload,\n' +
    'Shake or press menu button for dev menu',
});

export default class App extends Component {
  
  teste() {
    console.log('obj', SunmiInnerPrinter);
    SunmiInnerPrinter.printString('test');
  }
  
  render() {
      return (
          <View>
              <Header
                  leftComponent={{ icon: 'menu', color: '#fff' }}
                  centerComponent={{ text: 'MY TITLE', style: { color: '#fff' } }}
                  rightComponent={{ icon: 'home', color: '#fff' }}
              />
              <Text style={styles.welcome}>
                  Welcome to React Native!
              </Text>
              <Text style={styles.instructions}>
                  To get started, edit App.js
              </Text>
              <Text style={styles.instructions}>
                  {instructions}
              </Text>
              <Button
                  raised
                  icon={{ name: 'cached' }}
                  title='Print!!!' onPress={()=>{this.teste() }} />
          </View>
    );
  }
}

const styles = StyleSheet.create({
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

**Comment by januslo** [comment]
Should have a reference to the module class, manpackage class missing should be a problem. May you try to install a other plugin to test whether your project is fine? Emmm, what RN version you are working with?


发自我的 iPhone

**Comment by juniorschroder** [comment]
I'm using ReactNative Elements. this uses react-native-vector-icons (npm i --save react-native-vector-icons && react-native link react-native-vector-icons) and works fine.

I'm current using "react-native": "0.55.4",

**Comment by januslo** [comment]
hmmmm. one more try， uninstall and remove current react-native-sunmi-inner-printer your installed, then  install the master branch via  https://github.com/januslo/react-native-sunmi-inner-printer.git and test again. I have no update the NPM repo for a century. sorry about that.

**Comment by januslo** [comment]
That is abnormal,  This error usually shows the link issue.  You may need to make sure plugin is linked successfully. 

发自我的 iPhone

**Comment by januslo** [comment]
@juniorschroder  I have tested my demo and works fine. after that, I published the v0.1.6 to the NPM with the code in the master.

**Comment by juniorschroder** [comment]
@januslo, that's too weird!

I unlinked U package from my project and uninstall on npm, than I install and linked again and same thing, import SunmiInnerPrinter from 'react-native-sunmi-inner-printer'; and SunmiInnerPrinter is undefined!

So i did a test here, in U npm package, I've edited index.js and below const SunmiInnerPrinter = NativeModules.SunmiInnerPrinter; i put and console.log and ok, it's a Object, but after 
module.exports = SunmiInnerPrinter;
in my App this turns undefined.

So, to make this work I'm using on my App this module directly from react-native NativeModules
let test = NativeModules.SunmiInnerPrinter;

this works fine!

Thanks for attention!
